RealmSwift/LinkingObjects.swift

```diff
@@ -323,6 +323,130 @@ import Realm
     }
 
+    /**
+     Registers a block to be called each time the collection changes.
+
+     The block will be asynchronously called with the initial results, and then called again after each write
+     transaction which changes either any of the objects in the collection, or which objects are in the collection.
+
+     The `change` parameter that is passed to the block reports, in the form of indices within the collection, which of
+     the objects were added, removed, or modified during each write transaction. See the `RealmCollectionChange`
+     documentation for more information on the change information supplied and an example of how to use it to update a
+     `UITableView`.
+
+     At the time when the block is called, the collection will be fully evaluated and up-to-date, and as long as you do
+     not perform a write transaction on the same thread or explicitly call `realm.refresh()`, accessing it will never
+     perform blocking work.
+
+     If no queue is given, notifications are delivered via the standard run loop, and so can't be delivered while the
+     run loop is blocked by other activity. If a queue is given, notifications are delivered to that queue instead. When
+     notifications can't be delivered instantly, multiple notifications may be coalesced into a single notification.
+     This can include the notification with the initial collection.
+
+     For example, the following code performs a write transaction immediately after adding the notification block, so
+     there is no opportunity for the initial notification to be delivered first. As a result, the initial notification
+     will reflect the state of the Realm after the write transaction.
+
+     ```swift
+     let results = realm.objects(Dog.self)
+     print("dogs.count: \(dogs?.count)") // => 0
+     let token = dogs.observe { changes in
+         switch changes {
+         case .initial(let dogs):
+             // Will print "dogs.count: 1"
+             print("dogs.count: \(dogs.count)")
+             break
+         case .update:
+             // Will not be hit in this example
+             break
+         case .error:
+             break
+         }
+     }
+     try! realm.write {
+         let dog = Dog()
+         dog.name = "Rex"
+         person.dogs.append(dog)
+     }
+     // end of run loop execution context
+     ```
+
+     If no key paths are given, the block will be executed on any insertion,
+     modification, or deletion for all object properties and the properties of
+     any nested, linked objects. If a key path or key paths are provided,
+     then the block will be called for changes which occur only on the
+     provided key paths. For example, if:
+     ```swift
+     class Dog: Object {
+         @Persisted var name: String
+         @Persisted var age: Int
+         @Persisted var toys: List<Toy>
+     }
+     // ...
+     let dogs = realm.objects(Dog.self)
```

Comment:
The example for each of these should actually use the type which is being documented.

Reply:
Generally, it should. In this case I had put line 383 because a couple lines later the docs mentions:
```
- If the above example observed the `["toys"]` key path,
- ...
- If the observed key path were `["toys.brand"]`,
```
I gave the major key path combinations their own bullet point, instead of writing each in the sample block. Writing everything in line was less compact.
Do you think I scrap this format and have the details inline with code? Or reword the detail?

Reply:
Ah got it. I see what I misunderstood.

Reply:
What do you think is the cleanest way to do this for linking objects? I'm only coming up with declaring two classes like:

```
     ```swift
     class Person: Object {
         @Persisted var dogs: List<Dog>
     }
     class Dog: Object {
         @Persisted var name: String
         @Persisted var age: Int
         @Persisted var toys: List<Toy>
         @Persisted(originProperty: "dogs")
         var handlers: LinkingObjects<Person>
     }
     // ...
     let dogs = person.dogs
     let token = dogs.observe(keyPaths: ["name"]) { changes in
```

Reply:
It technically doesn't require two types but that's probably clearer than a self-referencing type. Person.dogs should be the LinkingObjects and Dog.handlers should be the forward link in that example though so that the thing being observed is a LinkingObjects.